gno.land/pkg/sdk/vm/keeper_test.go

```diff
@@ -955,3 +957,142 @@ func Echo(cur realm, msg string){
 	depDeltaFoo := env.bankk.GetCoins(ctx, depAddrFoo).Sub(depFoo)
 	assert.True(t, depDeltaTest.Add(depDeltaFoo).IsEqual(msg2.MaxDeposit))
 }
+
+// TestVMKeeperDeterminism tests that multiple runs of the same operations
+// produce the same app hash, ensuring deterministic behavior.
+// See #4580 for context.
+func TestVMKeeperDeterminism(t *testing.T) {
+	// Determine number of runs based on -short flag
+	numRuns := 50
+	if testing.Short() {
+		numRuns = 5
+	}
```

Comment:
The chance of getting the same sequence of packages with 2 runs is already abysmally low ($\frac{1}{10!}$ = 0.000000276), so I'd say to just run it with 5.

Reply:
i'm fine enforcing 5, but i know that some people could prefer increasing to 100 :)

Reply:
switched to 5: bdd6656c9